tests/test_end_scenario_sizing.py

```diff
@@ -98,6 +92,117 @@ def solver_options(self):
 
         np.testing.assert_allclose(obj / 1.0e6, solution.objective_value)
 
+    def test_end_scenario_sizing_staged(self):
+        """
+        Check if the EndScenarioSizingStagedHIGHS workflow is behaving as expected. This is an
+        optimization done over a full year with timesteps of 5 days and hour timesteps for the peak
+        day.
+
+        Checks:
+        - Cyclic behaviour for ATES
+        - That buffer tank is only used on peak day
+        - demand matching
+        - Check if TCO goal included the desired cost components.
+
+        - Compare objective value of staged approach without staged approach
+
+        Missing:
+        - Link ATES t0 utilization to state of charge at end of year for optimizations over one
+        year.
+        """
+        import models.test_case_small_network_ates_buffer_optional_assets.src.run_ates as run_ates
+
+        base_folder = Path(run_ates.__file__).resolve().parent.parent
+
+        # This is an optimization done over a full year with timesteps of 5 days and hour timesteps
+        # for the peak day
+
+        solution_unstaged = run_optimization_problem(
+            EndScenarioSizingHIGHS,
+            base_folder=base_folder,
+            esdl_file_name="test_case_small_network_with_ates_with_buffer_all_optional.esdl",
+            esdl_parser=ESDLFileParser,
+            profile_reader=ProfileReaderFromFile,
+            input_timeseries_file="Warmte_test.csv",
+        )
+        solution_unstaged_2 = run_end_scenario_sizing(
+            EndScenarioSizingHIGHS,
+            staged_pipe_optimization=False,
+            base_folder=base_folder,
+            esdl_file_name="test_case_small_network_with_ates_with_buffer_all_optional.esdl",
+            esdl_parser=ESDLFileParser,
+            profile_reader=ProfileReaderFromFile,
+            input_timeseries_file="Warmte_test.csv",
+        )
+        # TODO: check staged approach, the pipe flow direction seems to give some unwanted flow
+        #  directions, because objective becomes 1% higher enventhough MIPGap becomes 0%, when
+        #  staged_pipe_optimization=False, the same occurs that the objective is lower
```

Comment:
this is probably minimum flow velocity difference between stage 1 and 2

Reply:
old TODO, I already fixed this.